react/features/connection-indicator/statsEmitter.js

```diff
@@ -0,0 +1,174 @@
+ * as the real id might not be set before listeners start subscribing to local
+ * user stat updates. So instead, this constant is used.
+ */
+const LOCAL_USER_ID = 'local';
+
+/**
+ * Contains all the callbacks to be notified when stats are updated.
+ *
+ * {
+ *     userId: Function[]
+ * }
+ */
+const subscribers = {};
+
+/**
+ * A singleton that acts as a pub/sub service for connection stat updates.
+ */
+const statsEmitter = {
+    /**
+     * Emit a stat update to all those listening for local stat updates. Will
+     * also update listeners of remote user stats of changes related to their
+     * stats.
+     *
+     * @param {Object} stats - Connection stats for the local user as provided
+     * by the library.
+     * @returns {void}
+     */
+    localStatsUpdated(stats) {
+        const { framerate, resolution } = stats;
+        const currentLocalUserId = APP.conference.getMyUserId();
+
+        // FIXME overwrites 'lib-jitsi-meet' internal object
+        // Why library internal objects are passed as event's args ?
+        stats.resolution = resolution[currentLocalUserId];
+        stats.framerate = framerate[currentLocalUserId];
+
+        this.remoteStatsUpdated(LOCAL_USER_ID, stats);
+
+        Object.keys(resolution).forEach(id => {
+            if (id === currentLocalUserId) {
+                return;
+            }
+
+            const resolutionValue = resolution[id];
+
+            if (resolutionValue) {
+                this.remoteStatsUpdated(id, {
+                    resolution: resolutionValue
+                });
+            }
+        });
+
+        Object.keys(framerate).forEach(id => {
+            if (APP.conference.isLocalId(id)) {
+                return;
+            }
+
+            const framerateValue = framerate[id];
+
+            if (framerateValue) {
+                this.remoteStatsUpdated(id, {
+                    framerate: framerateValue
+                });
+            }
+        });
+    },
+
+    /**
+     * Emit a stat update to all those listening for a specific user's
+     * connection stats.
+     *
+     * @param {string} id - The user id the stats are associated with.
+     * @param {Object} stats - New connection stats for the user.
+     * @returns {void}
+     */
+    remoteStatsUpdated(id, stats = {}) {
+        const callbacks = subscribers[id] || [];
+
+        callbacks.forEach(callback => {
+            callback(stats);
+        });
+    },
+
+    /**
+     * Have {@code statsEmitter} subscribe to stat updates from a given
+     * conference.
+     *
+     * @param {JitsiConference} conference - The conference for which
+     * {@code statsEmitter} should subscribe for stat updates.
+     * @returns {void}
+     */
+    startListeningForStats(conference) {
+        const { connectionQuality } = JitsiMeetJS.events;
+
+        conference.on(connectionQuality.LOCAL_STATS_UPDATED,
+            stats => this.localStatsUpdated(stats));
```

Comment:
I don't think that's possible. Local stats have been handled specially in that local stats get parsed for framerate and resolution updates for remote users, so localStatsUpdated ends up emitting out updates for local and remote stats.